### DISCUSSION TITLE
uploading subscribers via CSV file for bulk provisioning

### DISCUSSION THREAD

**Discussion by pkapatsos-blinq** [magma/magma#6391] [Q&A] [answered]
Hi can anyone share what fields/Header info are needed in the CSV file to upload a bulk of CPEs to magma please, this link does not state this info or share a file to use.
https://docs.magmacore.org/docs/nms/subscriber#uploading-a-subscriber-csv-file
If you have a file to share with examples that would be much appreciated.

**Comment by karthiksubraveti** [comment]
https://docs.magmacore.org/docs/nms/subscriber#adding-new-subscriber

The bulk provisioning page opens up to an empty table. Rows can be added by clicking on the 'add row' button on the right. Each row contains columns including Subscriber Name, IMSI, Auth Key, Auth OPC, Service state, Data Plan, Active APN. auth key and auth_opc should be specified as hex. The service state can be either active or inactive. The data plan and active APN can be chosen based on what is available.

For example, John, IMSI001010000000079,<auth_key>,<auth_opc>,ACTIVE,default,oai_ipv4

**Comment by pkapatsos-blinq** [comment]
Thanks Kartik but my ask is on building a CSV file to import subscribers in bulk using ,
![image](https://user-images.githubusercontent.com/61711855/116246935-66d25680-a738-11eb-8118-bdadccfa2df2.png)
can you share a file ? or how the fields are for 2 users so we can build ? in your example this is one line, where does the 2nd sub go etc.
We need to import 50-100 Subs for testing and doing them one a t a time is ok for 10 or less  more than that is a bit annoying.

**Comment by karthiksubraveti** [comment, marked as answer]
[add_sub11.csv](https://github.com/magma/magma/files/6386630/add_sub11.csv)
Shared a sample subscribers file here

**Reply by pkapatsos-blinq** [reply]
Thank you for this.
